rpc, test: allow multiple data outputs in `createrawtransaction` & `createpsbt`

yuvicc opened the pull request:
Since #32406 is merged we can remove the check for multiple data outputs in `createrawtransaction` and `createpsbt` rpc.

**Keeping this in draft and review is currently not required as this is in milestone 31.0**

addresses: #32478

Dorex45 reviewed (commented):
The removal of the duplicate "data" key check simplifies the code path for handling data outputs. However, this means that if multiple "data" fields are present in the outputs object, the last one will silently override any previous entries without raising an error. Was this change intended to allow multiple "data" outputs, or is it assumed that input validation elsewhere will prevent duplicates? If not, could you clarify the motivation behind removing this check? Thanks!

Dorex45 commented:
> > If not, could you clarify the motivation behind removing this check? Thanks!
> 
> see #32478, we have policy change merged to remove data outputs size. So this check should be removed from the rpc's as well!

Thanks for the pointer to #32478 , that clears it up!
Since multiple data outputs are now valid, would it make sense to reflect that in the RPC help text or release notes for clarity? Also, adding a test with multiple data entries could help future-proof this change.
Nice cleanup overall!

luke-jr commented:
Concept NACK. Stop trying to enable spam.

yuvicc committed:
remove check from rawtransaction_util for duplicate data

test: remove assertions for duplicate data parameter from rpc_rawtransaction & wallet_bumpfee

yuvicc committed:
test: update check for multiple data outputs

sedited commented:
We're a few months on, and there still seems to be no interest in this PR. Maybe close it again @yuvicc ?

yuvicc commented:
Closing as there is no enough interest for this change.
